src/optimize.jl

```diff
@@ -0,0 +1,122 @@
+#######################################################
+# training loop for variational objectives 
+#######################################################
+function pm_next!(pm, stats::NamedTuple)
+    return ProgressMeter.next!(pm; showvalues=[tuple(s...) for s in pairs(stats)])
+end
+
+_wrap_in_DI_context(args) = DifferentiationInterface.Constant.([args...]) 
```

Comment:
given that this is just a one liner, I don't think extra function provides many benefits, how about using `DifferentiationInterface.Constant` directly?

Reply:
The broadcasting and splatting should probably also be avoided (regardless of whether it appears in a separate function or inline):
```suggestion
_wrap_in_DI_context(args) = map(DifferentiationInterface.Constant, args)
```